src/services/websockets/errors.ts

```diff
@@ -4,24 +4,43 @@
  */
 export abstract class WebsocketError extends Error {}
 
+/**
+ * Available websocket error names
+ *
+ * @example
+ *  ```
+ *    if (err.name === Websocket.ErrorNames.AccessDenied) {
+ *      // handle access denied
+ *    }
+ *  ```
+ */
+export enum ErrorNames {
+  AccessDenied = 'ACCESS_DENIED',
+  BadRequest = 'BAD_REQUEST',
+  NotFound = 'NOT_FOUND',
+  ServerError = 'SERVER_ERROR',
+}
+
+export type ErrorName = ErrorNames | `${ErrorNames}`;
+
 export class AccessDeniedError extends WebsocketError {
-  name = 'ACCESS_DENIED';
+  name: `${ErrorNames.AccessDenied}` = ErrorNames.AccessDenied;
```

Comment:
Curiosity: you want the `name` prop to have the literal type `'ACCESS_DENIED'` instead of the `ErrorNames` type (enum) ?

Reply:
Yes, this is what would discriminate this specific error type